cli/command/cli.go

```diff
@@ -82,6 +82,16 @@ type DockerCli struct {
 	dockerEndpoint     docker.Endpoint
 	contextStoreConfig store.Config
 	initTimeout        time.Duration
+
+	// baseCtx is the base context used for internal operations
+	// In the future this may be replaced by explicitly passing a context to functions that need it.
+	baseCtx context.Context
+}
+
+// WithContext sets the base context for the cli
+// This is used internally for operations may require a context to propagate tracing.
+func (cli *DockerCli) WithContext(ctx context.Context) {
+	cli.baseCtx = ctx
 }
 
```

Comment:
I'm moving this one to the OTEL commit; it's not used yet in this commit, and I want to look if this should be a `DockerCliOption` instead (also see https://github.com/docker/cli/pull/4387)